test/e2e/aadpodidentity_test.go

```diff
@@ -118,8 +114,8 @@ var _ = Describe("Kubernetes cluster using aad-pod-identity", func() {
 	AfterEach(func() {
 		fmt.Println("\nTearing down the test environment...")
 		if CurrentGinkgoTestDescription().Failed {
-			noCleanup = true
-			fmt.Println("Test failed. No cleanup performed")
+			fmt.Printf("Test failed. Collecting debugging information.")
+			collectDebuggingInfo()
```

Comment:
If there is an error with the current test, shouldn't we collect the debug info and still continue to do the cleanup? If we return, then we won't cleanup the assigned identities and the identity validator pods. We should get debug info and continue with the rest of `AfterEach`